### ISSUE TITLE
Is it possible to checkpoint scalar value?

### ISSUE THREAD

**Issue by Wongboo** [huggingface/accelerate#2738]
From accelerate document:
"By using [register_for_checkpointing()](https://huggingface.co/docs/accelerate/v0.29.3/en/package_reference/accelerator#accelerate.Accelerator.register_for_checkpointing), you can register custom objects to be automatically stored or loaded from the two prior functions, so long as the object has a state_dict and a load_state_dict functionality. This could include objects such as a learning rate scheduler."
Is it possible to include scalar value in a checkpoint just like `pytorch`? such as epoch and step, etc. if so, [redundant lines to recover epoch](https://github.com/huggingface/transformers/blob/a0e77a1f6bdfcceccdc5618e8a01ee32ef47bfa8/examples/pytorch/language-modeling/run_clm_no_trainer.py#L597-L606) can be avoided.

**Comment by muellerzr** [comment]
Sure we can absolutely. If you'd like to expand our checkpointing example here in accelerate implementing that, we can look at upstreaming it further 🤗
